react/features/video-quality/subscriber.js

```diff
@@ -289,110 +415,77 @@ function _updateReceiverVideoConstraints({ getState }) {
-            });
+            receiverConstraints[sourceNameSignaling ? 'selectedSources' : 'selectedEndpoints']
+                = remoteScreenSharesSourceNames;
         }
 
+    // Stage view.
     } else {
-        receiverConstraints = {
-            constraints: {},
-            defaultConstraints: { 'maxHeight': VIDEO_QUALITY_LEVELS.NONE },
-            lastN,
-            onStageEndpoints: [],
-            selectedEndpoints: []
-        };
-
-        // Tile view.
-        if (shouldDisplayTileView(state)) {
-            if (!visibleRemoteParticipants?.size) {
-                return;
-            }
+        if (!visibleRemoteTrackSourceNames?.length && !largeVideoSourceName && !activeParticipantsSources?.length) {
+            return;
+        }
 
-            visibleRemoteParticipants.forEach(participantId => {
-                receiverConstraints.constraints[participantId] = { 'maxHeight': maxFrameHeight };
+        if (visibleRemoteTrackSourceNames?.length) {
+            visibleRemoteTrackSourceNames.forEach(sourceName => {
+                receiverConstraints.constraints[sourceName] = { 'maxHeight': maxFrameHeightForVerticalFilmstrip };
             });
+        }
 
-            // Prioritize screenshare in tile view.
-            remoteScreenShares?.length && (receiverConstraints.selectedEndpoints = remoteScreenShares);
-
-        // Stage view.
-        } else {
-            if (!visibleRemoteParticipants?.size && !largeVideoParticipantId) {
-                return;
-            }
+        if (getCurrentLayout(state) === LAYOUTS.STAGE_FILMSTRIP_VIEW && activeParticipantsSources.length > 0) {
+            const onStageSources = [ ...activeParticipantsSources ];
 
-            if (visibleRemoteParticipants?.size > 0) {
-                const qualityLevel = getVideoQualityForResizableFilmstripThumbnails(state);
-                const stageParticipantsLevel = getVideoQualityForStageThumbnails(state);
+            activeParticipantsSources.forEach(sourceName => {
+                const isScreenSharing = remoteScreenShares.includes(sourceName);
+                const quality
+                    = isScreenSharing && preferredVideoQuality >= MAX_VIDEO_QUALITY
+                        ? VIDEO_QUALITY_UNLIMITED : maxFrameHeightForStageFilmstrip;
 
-                visibleRemoteParticipants.forEach(participantId => {
-                    const isStageParticipant = activeParticipantsIds.find(id => id === participantId);
-                    const quality = Math.min(maxFrameHeight, isStageParticipant
-                        ? stageParticipantsLevel : qualityLevel);
+                receiverConstraints.constraints[sourceName] = { 'maxHeight': quality };
+            });
 
-                    receiverConstraints.constraints[participantId] = { 'maxHeight': quality };
-                });
+            if (screenshareFilmstripParticipantId) {
```

Comment:
Same question as above for `screenshareFilmstripParticipantId` ? Isn't it already included in the visibleRemoteTrackSourceNames ?

Reply:
Same answer: it could be but I don't think this is an issue.